RFC/0001-graph-display.md

```diff
@@ -0,0 +1,241 @@
+to select the correct QML file to display the node after dispatching
+on the `type`.
+
+### 3.3 EdgeInterface
+
+Each edge implements five properties to do describe the spline
+connecting the nodes.  The `start` and `end` properties describe the
+exact starting and ending positions of the spline.  The `startHandle`
+and `endHandle` are the positions of the spline handels for the
+corresponding endpoints.  Finally, the `style` gives an index of the
+type of line being drawn.  For example, curves containing `int` values
+might have an index of 1 and `string` values an index of 2.  This
+provides the View with the information to display different edges
+while still deferring to the View about the exact display
+(e.g. allowing different colours to be chosen in a dark mode).
+
+### 3.4 Implementation
+
+In the actual QML, we will create a `GraphNodeView` that takes two
+parameters.  First first parameter is the `AbstractGraphModel` that
+contains the graph information.  The second is a delegate that accepts
+the `NodeInterface` and returns a group of widgets.  This delegate
+will *not* be responsible for adding the title bar to the node or
+drawing the node border.
+
+The implementation of this QML will be two `Repeaters`.  The first
+will iterate over the `nodes`, create the header and border, and use
+the provided delegate to populate the nodes.  The second will iterate
+over the `edges` and draw the splines.
+
+
+## 4 Metrics & Dashboards
+
+The final implementation will need to provide the following abilities:
+
+- Add new nodes of any type
+- Drag nodes to a new position
+- Draw connections between nodes with the mouse
+- Delete connections
+- Auto-arrange nodes
+- Display the nodes at multiple zoom levels
+- Scroll around the node environment
+
+None of these actions should leave the graph in an invalid state.
+
+## 5 Drawbacks
+
+- This project provides a significant time sink for at least one developer.
+
+- Even once the general library is written, more developer time will
+  be needed to write the Dissolve specific uses of the library
+
+
+## 6 Alternatives
+
+[Qt Node Editor](https://github.com/paceholder/nodeeditor) does not
+natively support QML and thus will require significant work to provide
+a consistent experience with the rest of the application.
+
+[QuickQanava](https://github.com/cneben/QuickQanava?tab=readme-ov-file)
+requires the construction and maintenance of an explicit graph object,
+as opposed to providing an abstract interface that we can build our
+own model again.  Additionally, at the time of this writing,
+QuickQanava does not build on Mac or Linux
+
+## 7 Potential Impact and Dependencies
+
+This could have a significant impact on the structure of the input
+file, since it represents a major change in the state of the data.
+Furthermore, this may be a significant break change, as it would not
+be a trivial matter to convert existing, linear code into the new
```

Comment:
This work would really represent a huge change - Dissolve 2 - and one which would break basically every existing input file.  It might be possible to write some kind of importer for 1.X TOML input, but it may be difficult and perhaps not worth the effort.  I would be happy to accept the fact that Dissolve 2 is a new starting point for everyone's simulations. One might argue there is little value to the user in importing an old simulation, since they will need to get used to the new workflows we'll be presenting.